### ISSUE TITLE
Error with choice operator when using vespa-documentgen-plugin

### ISSUE THREAD

**Issue by danitico** [vespa-engine/vespa#26596]
**Describe the bug**
I'm trying to setup a field that is populated with a default value when another field is null. The schema is the following:
```
field date type int {
   indexing: attribute | summary
}

field dateOldest type int {
    indexing: input (date || 99999999) | attribute
}
```

But when generating document classes with the plugin, I encounter the following error:

```
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:208)
    at com.yahoo.vespa.indexinglanguage.expressions.InputExpression.<init> (InputExpression.java:24)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.inputExp (IndexingParser.java:934)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.value (IndexingParser.java:401)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.math (IndexingParser.java:272)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.nonChoiceExpression (IndexingParser.java:249)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.expression (IndexingParser.java:227)
    at com.yahoo.vespa.indexinglanguage.parser.IndexingParser.statement (IndexingParser.java:204)
    at com.yahoo.vespa.indexinglanguage.ScriptParser.lambda$parseStatement$2 (ScriptParser.java:29)
    at com.yahoo.vespa.indexinglanguage.ScriptParser.parse (ScriptParser.java:51)
    at com.yahoo.vespa.indexinglanguage.ScriptParser.parseStatement (ScriptParser.java:27)
    at com.yahoo.vespa.indexinglanguage.expressions.StatementExpression.newInstance (StatementExpression.java:106)
    at com.yahoo.schema.fieldoperation.IndexingOperation.fromStream (IndexingOperation.java:51)
    at com.yahoo.schema.parser.SchemaParser.newIndexingOperation (SchemaParser.java:88)
    at com.yahoo.schema.parser.SchemaParser.newIndexingOperation (SchemaParser.java:73)
    at com.yahoo.schema.parser.SchemaParser.indexingOperation (SchemaParser.java:1786)
    at com.yahoo.schema.parser.SchemaParser.indexing (SchemaParser.java:1771)
    at com.yahoo.schema.parser.SchemaParser.fieldBody (SchemaParser.java:1664)
    at com.yahoo.schema.parser.SchemaParser.field (SchemaParser.java:842)
    at com.yahoo.schema.parser.SchemaParser.fieldOutsideDoc (SchemaParser.java:795)
    at com.yahoo.schema.parser.SchemaParser.rootSchemaItem (SchemaParser.java:288)
    at com.yahoo.schema.parser.SchemaParser.rootSchema (SchemaParser.java:218)
    at com.yahoo.schema.parser.SchemaParser.schema (SchemaParser.java:141)
    at com.yahoo.schema.parser.IntermediateCollection.addSchemaFromString (IntermediateCollection.java:49)
    at com.yahoo.schema.parser.IntermediateCollection.addSchemaFromFile (IntermediateCollection.java:113)
    at com.yahoo.schema.ApplicationBuilder.addSchemaFile (ApplicationBuilder.java:152)
    at com.yahoo.vespa.DocumentGenMojo.buildSearches (DocumentGenMojo.java:135)
    at com.yahoo.vespa.DocumentGenMojo.execute (DocumentGenMojo.java:106)
    at com.yahoo.vespa.DocumentGenMojo.execute (DocumentGenMojo.java:963)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**To Reproduce**
1. Have a schema with those two fields
2. Run the vespa documentgen plugin
3. See error

**Expected behavior**
The plugin should finish correctly.

**Environment (please complete the following information):**
 - OS: macOS
 - Infrastructure:self-hosted
 - Versions Ventura 13.2.1

**Vespa version**
8.144.19

**Comment by jobergum** [comment]
Does this correct syntax work?
```
field dateOldest type int {
    indexing: (input date || 99999999) | attribute
}
```

**Comment by bratseth** [comment]
These error messages are not very helpful, I've improved that in https://github.com/vespa-engine/vespa/pull/26666

You need to declare the "dateOldest" field outside the document block, since it takes the value of a document field instead of being one itself.

```
schema {
    document ... {
        ....
    }
    field dateOldest type int {
        indexing: (input date || 99999999) | attribute
    }
}
```

**Comment by danitico** [comment]
Hi @bratseth !  I had it outside the document block and I still have the same problem as in my last message.

**Comment by bratseth** [comment]
Ok, then you need to wait for https://github.com/vespa-engine/vespa/pull/26666 to be released. Unfortunately, we're not releasing during easter so that won't be until next week.

**Comment by danitico** [comment]
Hi again @bratseth ! I'm using the new deployed version (8.149.36) and I now have a different error:

```
For schema 'docs', field 'dateOldest': For expression 'input date || 99999999': Expected any output, got null.
```

**Comment by bratseth** [comment]
Sorry, this is fixed properly in 8.153.19. This time I added system testing ...
